### PR TITLE
feat: move all ldf config to be in xml

### PR DESCRIPTION
**Pull request by aronwk-aaron**
cleanup dev-tribute.xml
add comments to atm.xml
remove custom script tag in favor of ldfconfig for it
Add default client script to be defined in xml rather than a hardcoded special case
Use generic config option for random spawning rather than specifically teleport

Tested that everything loads still
Tested that things talk when they should
Tested that avery still teleports

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
converted all xml files from spaces to tabs as well
Ignore whitespace when reviewing those

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [inline dGame/dUtilities/VanityUtilities.cpp:256]: try parse here please

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [inline dGame/dUtilities/VanityUtilities.cpp:291]: try parse here please, though given its a block of these a general try and catch may work better for reporting on a failed to parse value.

**Review by EmosewaMC — COMMENTED** [review]
sto* questions

File: dGame/dUtilities/VanityUtilities.cpp
Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp:256]: Ah yeah, crap. Forgot to replace all of those

File: dGame/Entity.cpp
Comment (jadebenn) [inline dGame/Entity.cpp:2205]: Not really an issue, but I think your text editor isn't set to end files with a newline.

File: dGame/dUtilities/VanityUtilities.cpp
Comment (jadebenn) [inline dGame/dUtilities/VanityUtilities.cpp:27]: Nit: Should we be using the member naming convention outside a class?

File: dGame/dUtilities/VanityUtilities.cpp
Comment (jadebenn) [inline dGame/dUtilities/VanityUtilities.cpp:49]: A bit confused by the heap allocations here. Doesn't seem _wrong_, I just don't get it.

**Review by jadebenn — DISMISSED** [review]

File: dGame/Entity.cpp
Comment (aronwk-aaron) [reply dGame/Entity.cpp:2205]: I switch between python and C++ a bunch, so I leave autoformatting off unless I'm specifically doing formatting passes. newlines at eof's are something I miss when coding manually

File: dGame/dUtilities/VanityUtilities.cpp
Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp:49]: If there's a better way to do it, lmk. That's just how it is.

File: dGame/dUtilities/VanityUtilities.cpp
Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp:124]: no clue, I was asked to change this to be like this

File: dGame/dUtilities/VanityUtilities.cpp
Comment (jadebenn) [reply dGame/dUtilities/VanityUtilities.cpp:49]: Not an issue with this PR even if it is an issue (which I'm not sure of). If I have time later, I'll look into it myself, but I'll spin up a new PR in that case. Was just kind of hoping you had an easy explanation for me. 😅

File: dGame/dUtilities/VanityUtilities.cpp
Comment (jadebenn) [reply dGame/dUtilities/VanityUtilities.cpp:124]: I'm not necessarily opposed, but it'd be good to understand the rationale.

File: dGame/dUtilities/VanityUtilities.cpp
Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp:27]: no clue, as below, I was instructed on these changes by emo

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [reply dGame/dUtilities/VanityUtilities.cpp:49]: Thr heap allocation is necessary as ldf in dlu is designed poorly and hard to discern ownership.

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [reply dGame/dUtilities/VanityUtilities.cpp:124]: These public functions were never actually used in a public manner, and were not designed to, so they shouldn't have been public and likely were just because

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [inline dGame/dUtilities/VanityUtilities.cpp:211]: Has_value call can be removed.  Optional is implicitly convertible to bool

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [inline dGame/dUtilities/VanityUtilities.cpp:286]: Same as above

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [inline dGame/dUtilities/VanityUtilities.cpp:259]: Consider not spawning the object if it is of type lot null

Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp:259]: will do

**Review by Jettford — APPROVED** [review]
Approved on resolution of Emo's point.

**Review by EmosewaMC — APPROVED** [review]
Pending ci

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [reply dGame/dUtilities/VanityUtilities.cpp:27]: Member naming should be removed here to jades point, they aren't members anymore

Comment (EmosewaMC) [reply dGame/dUtilities/VanityUtilities.cpp:27]: But if u want to jeep the cl small, feel free to push it back

Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp:27]: ah missed what this mean, changed

**Comment by aronwk-aaron** [comment]
(tweaking some more vanity xml stuff currently in discord)

**Comment by aronwk-aaron** [comment]
PR done, wont touch it anymore besides feedback

**Comment by jadebenn** [comment]
> PR done, wont touch it anymore besides feedback

I'm afraid the merge conflict says otherwise 😜

File: dGame/dUtilities/VanityUtilities.cpp
Comment (jadebenn) [inline dGame/dUtilities/VanityUtilities.cpp:27]: Lgtm, incidentally. Would like to see the former members here uncapitalized (i.e. objects instead of Objects) but it's no big deal and I'll approve once the merge conflict is sorted

**Review by jadebenn — APPROVED** [review]

File: dGame/dUtilities/VanityUtilities.cpp
Comment (EmosewaMC) [reply dGame/dUtilities/VanityUtilities.cpp]: I think its good to keep file scope variables with this syntax so you know what you may be referring to at a glance (file scope container vs local scope)

Comment (aronwk-aaron) [reply dGame/dUtilities/VanityUtilities.cpp]: Want me to change it back then?

Comment (EmosewaMC) [reply dGame/dUtilities/VanityUtilities.cpp]: Not worth blocking further.  Though I believe pascal case for file scope is better.

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**
